### PR TITLE
Add `docs` requirements to `pyproject.toml`

### PR DESCRIPTION
**Pull request by jakirkham**
Instead of having a separate requirements file for RTD, fold these requirements into `pyproject.toml` as is done with Numcodecs.

### CODE REVIEW

**Comment by jakirkham** [comment]
Toggling for RTD. Looks like [the last build]( https://readthedocs.org/projects/zarr/builds/21553600/ ) was unable to fetch the PR

**Comment by jakirkham** [comment]
Weird RTD still [can't fetch the PR]( https://readthedocs.org/projects/zarr/builds/21553600/ )

**Comment by jakirkham** [comment]
Maybe this is upstream issue ( https://github.com/readthedocs/readthedocs.org/issues/10601 ). Not sure why it started happening just now. Numcodecs didn't have the same issue

**Comment by jakirkham** [comment]
Well at least RTD is reporting the status now, which it wasn't before. Let's see what upstream says

File: pyproject.toml [hunk 1]
```diff
@@ -43,6 +43,16 @@ jupyter = [
...
 ]
+docs = [
+    'sphinx',
+    'sphinx_design',
+    'sphinx-issues',
+    'sphinx-copybutton',
+    'sphinx-rtd-theme',
+    'pydata-sphinx-theme',
+    'numpydoc',
+    'numcodecs[msgpack]',
```
Comment (jakirkham) [inline pyproject.toml:54]: Since `msgpack` is a `numcodecs` requirement, specify it via `numcodecs`' extra requirement

**Comment by jakirkham** [comment]
@joshmoore could you please review? This is some follow on work from issue ( https://github.com/zarr-developers/zarr-python/issues/1492 )

**Review by joshmoore — APPROVED** [review]
:heart: More of this goodness!

**Pull request merged**